### PR TITLE
CMake: add GT_ENABLE_MPI_CUDA_TESTS option

### PR DESCRIPTION
**Pull request by havogt**
which allows to disable the tests which require a CUDA-aware MPI implementation.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
I thought it is a good idea, now I don't remember why. Probably it was not a good idea then. ;)

**Comment by havogt** [comment]
launch jenkins

File: cmake/public/gridtools_setup_targets.cmake [hunk 1]
```diff
@@ -228,7 +228,8 @@ macro(_gt_setup_targets _config_mode clang_cuda_mode)
         list(APPEND GT_BACKENDS cuda)
         list(APPEND GT_ICO_BACKENDS cuda)
 
-        if(MPI_CXX_FOUND)
+        option(GT_GCL_GPU "Disable if your MPI implementation is not CUDA-aware" ON)
```
Comment (anstaf) [inline cmake/public/gridtools_setup_targets.cmake:231]: I think this option should appear when both `MPI` and `CUDA` are found or it should be always available.

Comment (havogt) [reply cmake/public/gridtools_setup_targets.cmake]: You are right. I am doing too many things at the same time :(

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**